internal/client/http_client.go

```diff
@@ -38,6 +38,15 @@ func NewHTTPClient(config *proxy.TLSConfig) (*TanzuMissionControl, error) {
 		return nil, err
 	}
 
+	return newHTTPClient(httpClient), nil
+}
+
+// NewHTTPClientWithDefaultTransport is intended primarily for testing, as httpmock requires a default transport object be used.
+func NewHTTPClientWithDefaultTransport() *TanzuMissionControl {
```

Comment:
if it is going to be used only for tests, can this be added only in the test files and not in the main code flow ?